### PR TITLE
Xml formatter root and object tags

### PR DESCRIPTION
**Pull request by samdark**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #12419

### CODE REVIEW

File: framework/web/XmlResponseFormatter.php [hunk 1]
```diff
@@ -37,7 +37,7 @@ class XmlResponseFormatter extends Component implements ResponseFormatterInterfa
      */
     public $encoding;
     /**
-     * @var string the name of the root element.
+     * @var string the name of the root element or null if no root tag should be added.
```
Comment (cebe) [inline framework/web/XmlResponseFormatter.php]: in Html helper we use false for no tag. In general we could also allow it to be empty(). as `<>` would not be a valid xml tag anyway.

File: framework/web/XmlResponseFormatter.php [hunk 2]
```diff
@@ -50,7 +50,11 @@ class XmlResponseFormatter extends Component implements ResponseFormatterInterfa
      * @since 2.0.7
      */
     public $useTraversableAsArray = true;
-
+    /**
+     * @var bool if object tags should be added
+     * @since 2.0.10
```
Comment (cebe) [inline framework/web/XmlResponseFormatter.php]: 2.0.11.

**Review by cebe — CHANGES_REQUESTED** [review]

**Review by cebe — APPROVED** [review]

**Review by SilverFire — APPROVED** [review]

File: framework/web/XmlResponseFormatter.php [hunk 3]
```diff
@@ -50,7 +50,11 @@ class XmlResponseFormatter extends Component implements ResponseFormatterInterfa
      * @since 2.0.7
      */
     public $useTraversableAsArray = true;
-
+    /**
+     * @var bool if object tags should be added
+     * @since 2.0.11
+     */
+    public $objectTags = true;
```
Comment (klimov-paul) [inline framework/web/XmlResponseFormatter.php]: `objectTags` does not sound appropriate for a boolean field.
Possible names could be:
- `createObjectTags`
- `useObjectTags`
- `addObjectTags`

Comment (samdark) [inline framework/web/XmlResponseFormatter.php]: Yes. `useObjectTags` sounds better...

**Pull request merged**